butte-build/src/codegen.rs

```diff
@@ -0,0 +1,559 @@
+use crate::types::*;
+#[cfg(test)]
+use crate::{error::Error, field, table};
+use heck::SnakeCase;
+use proc_macro2::TokenStream;
+use quote::format_ident;
+use quote::quote;
+use quote::ToTokens;
+use std::convert::TryInto;
+use std::io::Read;
+
+#[cfg(test)]
```

Comment:
Make a `mod tests` instead, https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html

Reply:
I did this because I want to have the tests near the implementation, so that it's easy to see what things have a test vs not.

Reply:
Ended up moving everything into submodules. I didn't realize that `#[test]` doesn't imply `#[cfg(test]`.